src/page.ts

```diff
@@ -286,6 +286,31 @@ export class Page extends EventTarget {
     return this.#celestial;
   }
 
+  /**
+   * Provide credentials for HTTP authentication.
+   *
+   * @example
+   * ```ts
+   * await page.authenticate({ 'username': username, 'password': password });
+   * ```
+   */
+  authenticate(
+    { username, password }: { username: string; password: string },
+  ): Promise<void> {
+    function base64encoded(s: string) {
+      const codeUnits = new Uint16Array(s.length);
+      for (let i = 0; i < codeUnits.length; i++) {
+        codeUnits[i] = s.charCodeAt(i);
+      }
+      return btoa(String.fromCharCode(...new Uint8Array(codeUnits.buffer)));
+    }
+
+    const auth = base64encoded(`${username}:${password}`);
+    return this.#celestial.Network.setExtraHTTPHeaders({
+      headers: { "Authorization": `Basic ${auth}` },
+    });
+  }
+
```

Comment:
Could we have a test for this somehow?

Reply:
> Could we have a test for this somehow?

test failed. Any ideas about this?
Maybe I should take a look at puppeteer's source code in my spare time